src/notation/view/percussionpanel/percussionpanelpadmodel.h

```diff
@@ -36,6 +39,8 @@ class PercussionPanelPadModel : public QObject, public muse::async::Asyncable
     Q_PROPERTY(QString keyboardShortcut READ keyboardShortcut NOTIFY keyboardShortcutChanged)
     Q_PROPERTY(QString midiNote READ midiNote NOTIFY midiNoteChanged)
 
+    Q_PROPERTY(PaintedEngravingItem * notationPreview READ notationPreview WRITE setNotationPreview NOTIFY notationPreviewChanged)
```

Comment:
You could consider doing this the other way around: making `PaintedEngravingItem` a bit smarter, renaming it to `PercussionPanelPadNotationPreview`, and telling it about the `PercussionPanelPadModel`. Then let the `PercussionPanelPadNotationPreview` retrieve the element, based on info from `PercussionPanelPadModel`. 

That may mean that `PercussionPanelPadModel` needs to become aware of the `Drumset*`; I think that's okay. 

If you take that a bit further, it would mean that `PercussionPanelPadModel` only needs a Drumset* and a pitch in its constructor (besides QObject parent). That would simplify `PercussionPanelPadListModel::resetLayout` quite a bit, because `PercussionPanelPadModel` takes over some responsibilities from there. I'm not sure if that's the way you want to go, but personally I would not be against it.

Reply:
One of the reasons I like the current approach is because, fundamentally, I think `PaintedEngravingItem` could be very useful beyond the percussion panel. For example, we have some future plans to revamp the “edit drumset” dialog and I’m pretty sure a component like this will come in handy over there!